spec/integration/transport/winrm_spec.rb

```diff
@@ -1126,7 +1128,9 @@ def stub_winrm_to_raise(klass, message)
         end
       end
 
-      it "returns a friendly stderr msg with puppet.bat missing", winrm: true do
+      # This test currently won't pass because all windows integration tests
+      # are run on the same VM, which has puppet-agent installed.
+      xit "returns a friendly stderr msg with puppet.bat missing", winrm: true do
```

Comment:
Mmk, it seems like the last thing to do is to update this tag to be `winrm_agentless` or something along those lines, and then run tests with that tag in a separate job inside the `.github/workflows/windows` file, then exclude that tag when running the Windows integration tests + re-enable these tests.